ci/setup-build-test-gdk.ps1

```diff
@@ -1,66 +1,85 @@
-		$this.test_results_dir = $test_results_dir
-		$this.tests_path = $tests_path
-		$this.run_with_spatial = $run_with_spatial
-	}
+    [ValidateNotNullOrEmpty()][string]$test_repo_url
+    [ValidateNotNullOrEmpty()][string]$test_repo_branch
+    [ValidateNotNullOrEmpty()][string]$test_repo_relative_uproject_path
+    [ValidateNotNullOrEmpty()][string]$test_repo_map
+    [ValidateNotNullOrEmpty()][string]$test_project_name
+    [ValidateNotNullOrEmpty()][string]$test_results_dir
+    [ValidateNotNullOrEmpty()][string]$tests_path
+    [ValidateNotNull()]       [string]$additional_gdk_options
+    [bool]                            $run_with_spatial
+
+    TestSuite([string] $test_repo_url, [string] $test_repo_branch, [string] $test_repo_relative_uproject_path, [string] $test_repo_map, [string] $test_project_name, [string] $test_results_dir, [string] $tests_path, [string] $additional_gdk_options, [bool] $run_with_spatial) {
+        $this.test_repo_url = $test_repo_url
+        $this.test_repo_branch = $test_repo_branch
+        $this.test_repo_relative_uproject_path = $test_repo_relative_uproject_path
+        $this.test_repo_map = $test_repo_map
+        $this.test_project_name = $test_project_name
+        $this.test_results_dir = $test_results_dir
+        $this.tests_path = $tests_path
+        $this.additional_gdk_options = $additional_gdk_options
+        $this.run_with_spatial = $run_with_spatial
+    }
 }
 
 [string] $test_repo_url = "git@github.com:improbable/UnrealGDKEngineNetTest.git"
 [string] $test_repo_relative_uproject_path = "Game\EngineNetTest.uproject"
 [string] $test_repo_map = "NetworkingMap"
 [string] $test_project_name = "NetworkTestProject"
 [string] $test_repo_branch = "master"
-[bool]   $slow_networking_tests = $False
+[string] $user_gdk_settings = ""
 
 # Allow overriding testing branch via environment variable
 if (Test-Path env:TEST_REPO_BRANCH) {
-	$test_repo_branch = $env:TEST_REPO_BRANCH
+    $test_repo_branch = $env:TEST_REPO_BRANCH
 }
-# Allow overriding running slow networking tests
-if (((Test-Path env:SLOW_NETWORKING_TESTS) -And ($env:SLOW_NETWORKING_TESTS -eq "true")) -Or ($env:NIGHTLY_BUILD -eq "true")) {
-	$slow_networking_tests = $True
+
+if (Test-Path env:GDK_SETTINGS) {
+    $user_gdk_settings = ";" + $env:GDK_SETTINGS
 }
 
-$tests = @(
-	[TestSuite]::new("$test_repo_url", "$test_repo_branch", "$test_repo_relative_uproject_path", "$test_repo_map", "$test_project_name", "FastTestResults", "SpatialGDK+/Game/SpatialNetworkingMap", $True)
-)
+$tests = @()
 
-if ($slow_networking_tests) {
-	$tests[0].tests_path += "+/Game/NetworkingMap"
-	$tests[0].test_results_dir = "SpatialTestResults"
-	$tests += [TestSuite]::new("$test_repo_url", "$test_repo_branch", "$test_repo_relative_uproject_path", "$test_repo_map", "$test_project_name", "VanillaTestResults", "/Game/NetworkingMap", $False)
+# If building all configurations, use the test gyms, since the network testing project only compiles for the Editor configs
```

Comment:
when should we use the test gyms vs network test project? would it make sense to always use the test gyms?

Reply:
Test gyms have these build configs enabled, so we use them for testing builds, but the network test project has the functional tests, so we use it for running more tests. The way to fix this in the future so we only use one project would be to move to the network test project and make it work with all configurations. Currently this is difficult, because the NetTest project relies on editor modules (maybe we could somehow change that), so will not run in these build configs.

Reply:
ok, got it. Thanks. :) Might be good to add a comment about what we test in either repo to the file